tests/integration/cmd_pref_config_test.go

```diff
@@ -234,25 +234,36 @@ var _ = Describe("odo preference and config command tests", func() {
 	})
 
 	Context("when viewing local config without logging into the OpenShift cluster", func() {
-		It("should list config successfully", func() {
-			helper.Cmd("odo", "create", "--s2i", "nodejs", "--git", "https://github.com/openshift/nodejs-ex", "--project", commonVar.Project, "--context", commonVar.Context).ShouldPass()
-			helper.Cmd("odo", "config", "set", "--env", "hello=world", "--context", commonVar.Context).ShouldPass()
-			kubeconfigOld := os.Getenv("KUBECONFIG")
-			os.Setenv("KUBECONFIG", "/no/such/path")
-			configValue := helper.Cmd("odo", "config", "view", "--context", commonVar.Context).ShouldPass().Out()
-			helper.MatchAllInOutput(configValue, []string{"hello", "world"})
-			os.Setenv("KUBECONFIG", kubeconfigOld)
+		var ocRunner helper.OcRunner
+		var token string
+		BeforeEach(func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+			helper.Cmd("odo", "create", "--s2i", "nodejs", "nodejs", "--project", commonVar.Project, "--context", commonVar.Context).ShouldPass()
+			ocRunner = helper.NewOcRunner("oc")
+			token = ocRunner.GetToken()
+			ocRunner.Logout()
 		})
-
-		It("should set config variable without logging in", func() {
-			helper.Cmd("odo", "create", "--s2i", "nodejs", "--project", commonVar.Project, "--context", commonVar.Context).ShouldPass()
-			kubeconfigOld := os.Getenv("KUBECONFIG")
-			os.Setenv("KUBECONFIG", "/no/such/path")
-			helper.Cmd("odo", "config", "set", "--force", "--context", commonVar.Context, "Name", "foobar").ShouldPass()
-			configValue := helper.Cmd("odo", "config", "view", "--context", commonVar.Context).ShouldPass().Out()
-			Expect(configValue).To(ContainSubstring("foobar"))
-			helper.Cmd("odo", "config", "unset", "--force", "--context", commonVar.Context, "Name").ShouldPass()
-			os.Setenv("KUBECONFIG", kubeconfigOld)
+		AfterEach(func() {
+			ocRunner.LoginUsingToken(token)
+		})
+		When("user is working with a devfile component", func() {
+			It("should set, list and delete config successfully", func() {
+				if helper.IsKubernetesCluster() {
+					Skip("skipping for kubernetes until we can figure out how to simulate logged out state there")
```

Comment:
Does this thing work fine when a user is logged in? Or this error is only bound to occur when a user is logged out, because if a user was able to successfully log in, then it means that the KUBECONFIG file is in place and is error-free?

Reply:
as long as whatever kubeconfig the user has provided is valid, it will work. Being `logged in` is not required